cd/bootstrap/cfn-templates/cicd-pipe-70-microsvc_pipeline.yaml

```diff
@@ -0,0 +1,745 @@
+              - iam:ListRoles
+              - cloudformation:Describe*
+              - cloudFormation:List*
+              - codecommit:List*
+              - codecommit:Get*
+              - codecommit:GitPull
+              - codecommit:UploadArchive
+              - codecommit:CancelUploadArchive
+              - codebuild:BatchGetBuilds
+              - codebuild:StartBuild
+              - cloudformation:CreateStack
+              - cloudformation:DeleteStack
+              - cloudformation:DescribeStacks
+              - cloudformation:UpdateStack
+              - cloudformation:CreateChangeSet
+              - cloudformation:DeleteChangeSet
+              - cloudformation:DescribeChangeSet
+              - cloudformation:ExecuteChangeSet
+              - cloudformation:SetStackPolicy
+              - cloudformation:ValidateTemplate
+              - iam:PassRole
+              - s3:ListAllMyBuckets
+              - s3:GetBucketLocation
+              - "ecr:*"
+              - "*"
+            Resource:
+              - "*"
+          -
+            Effect: Allow
+            Action:
+              - kms:Decrypt
+            Resource: !Ref CMKARN
+          -
+            Effect: Allow
+            Action:
+              - s3:PutObject
+              - s3:GetBucketPolicy
+              - s3:GetObject
+              - s3:ListBucket
+            Resource:
+              - !Sub arn:aws:s3:::${ArtifactBucket}/*
+              - !Sub arn:aws:s3:::${TemplateBucket}/*
+          -
+            Effect: Allow
+            Action:
+              - sts:AssumeRole
+            Resource:
+              - !Sub ${GithubAccessRole.Arn}
+              - !Sub arn:aws:iam::${DevAccount}:role/${ProjectName}-CdPipelineRole
+              - !Sub arn:aws:iam::${UatAccount}:role/${ProjectName}-CdPipelineRole
+              - !Sub arn:aws:iam::${ProdAccount}:role/${ProjectName}-CdPipelineRole
+
+  
+  # PIPELINE DEFINITION
+  #######################
+  MicroservicePipeline:
+    Type: AWS::CodePipeline::Pipeline
+    Properties:
+      RoleArn: !GetAtt PipeLineRole.Arn
+      Name: !Sub '${ProjectName}-${MicroserviceName}-Pipeline'
+      Stages:
+        # Download sources: image and files for CFN templates
+        - Name: Get_Sources
+          Actions:
+            # Get microservice image digest
+            - Name: Microservice_image_source
+              ActionTypeId:
+                Version: '1'
+                Owner: AWS
+                Category: Source
+                Provider: ECR
+              RunOrder: 1
+              Configuration:
+                RepositoryName: !Select [ 0, !Split [ ":", !Ref MicroserviceImageNameAndTag ] ]
+                ImageTag: !Select [ 1, !Split [ ":", !Ref MicroserviceImageNameAndTag ] ]
+              OutputArtifacts:
+                - Name: MicrosvcImageArtifact
+              Namespace: 'MicrosvcImageVars'
+            # Checkout infrastructures CFN templates from Github
+            - Name: Infra_form_Github
```

Comment:
Si potrebbe semplificare usando un Bucket S3 in cui la pipeline di Infra ha predisposto i template necessari al posto di riscaricare il repository?
Cosa cambia? Utilizzo dei template di infra come erano al momento del "deploy" e non del branch. Questo mi fa perdere la flessibilità di testare un template da un branch/versione diversa da quella dispiegata.

Reply:
Mi sembra che il bucket renda più difficile rispondere alla domanda "Cosa è stato usato e come mai?"
Il numero di versione di un file S3 è difficile da correlare a una storia. Un commit git è facilmente correlabile a una storia di sviluppo.